tests/main/dirs-not-shared-with-host/task.yaml

```diff
@@ -26,15 +27,16 @@ environment:
     DIRECTORY/alternatives: /etc/alternatives
     DIRECTORY/ssl: /etc/ssl
 execute: |
+    #shellcheck source=tests/lib/dirs.sh
     . "$TESTSLIB/dirs.sh"
     echo "We can check the inode number of $DIRECTORY"
-    host_inode="$(stat -c '%i' $DIRECTORY)"
+    host_inode=$(stat -c '%i' "$DIRECTORY")
     if [ -e $SNAP_MOUNT_DIR/core/current ]; then
```

Comment:
Don't we need to quote this? I.e. `if [ -e "$SNAP_MOUNT_DIR/..."` ? I.e. is shellcheck missing this?

Reply:
I've seen this in a couple of places. My gut feeling is that since shellcheck follows the sourced files it can prove that SNAP_MOUNT_DIR is a path that does not require quoting. Try this example:

```
#!/bin/bash

SNAP_MOUNT_PATH_GOOD=/foo/bar
SNAP_MOUNT_PATH_BAD=$(ls -1 . | head -1)

if [ -d $SNAP_MOUNT_PATH_GOOD/bar/bar ]; then
        echo "ok"
fi

if [ -d $SNAP_MOUNT_PATH_BAD/bar/bar ]; then
        echo "ok"
fi
```